ambry-mysql/src/main/java/com/github/ambry/mysql/MySqlDataAccessor.java

```diff
@@ -177,6 +181,10 @@ public synchronized Connection getDatabaseConnection(boolean needWritable) throw
 
     // If the active connection is good and it's the best endpoint, keep it.
     if (activeConnection != null && !isBetterEndpoint(bestEndpoint, connectedEndpoint)) {
+      if (config != null && !config.transactionIsolationLevel.equals(TransactionIsolationLevel.TRANSACTION_NONE)) {
+        activeConnection.setTransactionIsolation(
```

Comment:
May be we can set the transaction isolation level immediately after creation connection (after below code). In that way, we can have it just in one place? 

```
 Connection connection = mysqlDriver.connect(candidateEndpoint.getUrl(), credentials);
```

Reply:
yeah, we can do that, the reason I set to two place is due to if the connection is still active, we are still able to set the transaction isolation level, but since the connection will be closed eventually and go through the connectToBestAvailableEndpoint logic, so I guess it should be fine.